.github/workflows/install-slices.yaml

```diff
@@ -0,0 +1,121 @@
+  RELEASES: ${{ toJson('["ubuntu-20.04","ubuntu-22.04","ubuntu-23.10","ubuntu-24.04"]') }}
+
+jobs:
+  prepare-install:
+    runs-on: ubuntu-latest
+    name: "Prepare to install"
+    outputs:
+      install-all: ${{ steps.set-output.outputs.install_all }}
+      matrix: ${{ steps.set-output.outputs.matrix }}
+    steps:
+      - name: Set output
+        id: set-output
+        run: |
+          set -ex
+          if [[
+            "${{ github.base_ref || github.ref_name }}" == "main" ||
+            "${{ github.event_name }}" == "schedule"
+          ]]; then
+            echo "matrix={\"ref\":${{ env.RELEASES }},\"arch\":${{ env.ARCHES }}}" >> $GITHUB_OUTPUT
+            echo "install_all=true" >> $GITHUB_OUTPUT
+          else
+            echo "matrix={\"arch\":${{ env.ARCHES }}}" >> $GITHUB_OUTPUT
+          fi
+
+  # The "install" job tests the slices by installing them.
+  # It installs **all** slices if:
+  #   - inputs.install-all is true
+  #   - chisel.yaml is changed
+  #   - any slice definition files are deleted
+  #   - github workflows (and related files) are changed
+  # Otherwise, it installs only the slices from **added** and/or
+  # **modified** slice definition files.
+  # Please note that "change" is defined to encompass additions,
+  # modifications and deletions here.
+  install:
+    runs-on: ubuntu-latest
+    name: "Install"
+    needs: prepare-install
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.prepare-install.outputs.matrix) }}
+    env:
+      install-all: ${{ needs.prepare-install.outputs.install-all }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ matrix.ref }}
+
+      - name: Check changed paths
+        id: changed-paths
+        if: env.install-all != 'true'
+        uses: dorny/paths-filter@v3
+        with:
+          # ref: https://github.com/marketplace/actions/paths-changes-filter
+          filters: |
+            install-all:
+              - 'chisel.yaml'
+              - deleted: 'slices/**/*.yaml'
+              - '.github/**'
+            slices:
+              - added|modified: 'slices/**/*.yaml'
+          # Space delimited list usable as command-line argument list in
+          # Linux shell. If needed, it uses single or double quotes to
+          # wrap filename with unsafe characters.
+          list-files: shell
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+
+      - name: Install dependencies
+        run: |
+          go install github.com/canonical/chisel/cmd/chisel@latest
+          go install github.com/mikefarah/yq/v4@latest
+          wget -q https://raw.githubusercontent.com/canonical/chisel-releases/main/.github/scripts/install-slices
+          chmod +x install-slices
+
+      - name: Install all slices
+        if: |
+          env.install-all == 'true' ||
+          steps.changed-paths.outputs.install-all == 'true'
```

Comment:
does this work if `changed-paths` was not executed?

Reply:
If `env.install-all` is `true`, then the `changed-paths` step would not run resulting in an empty value for `steps.changed-paths.outputs.install-all` i.e. `steps.changed-paths.outputs.install-all=""`. Otherwise, if `env-install` is not `true`, the `changed-paths` step would be executed resulting in `true` or `false` values for the outputs of the job.

For this step, yeah, it does work because either `env.install-all` would be `true` or the `steps.changed-paths.outputs.install-all` would have some value `true` or `false`.

Reply:
Note that this has been changed in https://github.com/canonical/chisel-releases/pull/119/commits/71a897eea58ac17332275feb6b252f1fb5d0931d. The comparison is now being done inside the `run` field.